suricata/update/rule.py

```diff
@@ -310,7 +313,13 @@ def parse_fileobj(fileobj, group=None):
         try:
             rule = parse(buf, group)
             if rule:
-                rules.append(rule)
+                try:
+                    if rule["sid"]!=None:
+                        rules.append(rule)
+                    else:
+                        raise BadSidError("Sid cannot be of type null")
+                except BadSidError as err:
+                    logger.error("Failed to parse rule: %s: %s", buf.rstrip(), err)
```

Comment:
Was this extra nested try..except block required? Could you please try doing it in a cleaner way?